### PR TITLE
[WebGL2] Don't apply buffer offset twice in getBufferSubData

### PR DESCRIPTION
**Pull request by djg**
#### 747aaf24c4e99c2fcb28a00c8620072a834874a0
<pre>
[WebGL2] Don&apos;t apply buffer offset twice in getBufferSubData
<a href="https://bugs.webkit.org/show_bug.cgi?id=243566">https://bugs.webkit.org/show_bug.cgi?id=243566</a>
rdar://98154048

Reviewed by Kimmo Kinnunen.

* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::getBufferSubData):
baseAddress() already takes care of offset into ArrayBuffer.

Canonical link: <a href="https://commits.webkit.org/253175@main">https://commits.webkit.org/253175@main</a>
</pre>

### CODE REVIEW

File: LayoutTests/fast/canvas/webgl/webgl2-getbuffersubdata.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+<head>
+<script src="../../../resources/js-test-pre.js"></script>
+</head>
+<body>
+<canvas id="canvas" width="40" height="40"></canvas>
+<script>
+description("Make sure that reading from WebGL2 buffers into offset ArrayBufferView works as expected.");
+
+if (window.internals)
+	internals.settings.setWebGL2Enabled(true);
```
Comment (kkinnunen-apple) [inline LayoutTests/fast/canvas/webgl/webgl2-getbuffersubdata.html]: I think you could use 2 or 4 space indent, adding yet another indent convention may be detrimental.

**Review by kkinnunen-apple — APPROVED** [review]

File: LayoutTests/fast/canvas/webgl/webgl2-getbuffersubdata.html (hunk 1 above)
Comment (djg) [reply LayoutTests/fast/canvas/webgl/webgl2-getbuffersubdata.html]: Converted tabs to 4 spaces.

**Comment by kenrussell** [comment]
@djg would you be able to put up a pull request adding any needed new cases to for example https://github.com/KhronosGroup/WebGL/blob/main/sdk/tests/conformance2/buffers/get-buffer-sub-data.html or https://github.com/KhronosGroup/WebGL/blob/main/sdk/tests/conformance2/buffers/get-buffer-sub-data-validity.html ? Appreciate it - regression tests for browser issues are important. Thanks.

**Comment by djg** [comment]
> @djg would you be able to put up a pull request adding any needed new cases [...]

Will do.

**Pull request merged**